opensearchapi/api.document_test.go

```diff
@@ -0,0 +1,329 @@
+				Body:       strings.NewReader(`{ "doc": { "title": "Moneyball", "director": "Bennett", "year": "2012" } }`),
+				Source:     []string{"true"},
+			},
+			want: &opensearchapi.Response{
+				StatusCode: 200,
+				Header: http.Header{
+					"Content-Type": []string{"application/json; charset=UTF-8"},
+				},
+			},
+			wantBody: fmt.Sprintf(`{"_index":"%s","_id":"1","_version":2,"result":"noop","_shards":{"total":0,"successful":0,"failed":0},"_seq_no":2,"_primary_term":1,"get":{"_seq_no":2,"_primary_term":1,"found":true,"_source":{"title":"Moneyball","director":"Bennett","year":"2012"}}}`, index),
+			wantErr:  false,
+		},
+		{
+			name: "TestUpdateRequest_Do. Source Excludes",
+			r: opensearchapi.UpdateRequest{
+				Index:          index,
+				DocumentID:     "1",
+				Body:           strings.NewReader(`{ "doc": { "title": "Moneyball", "director": "Bennett", "year": "2012" } }`),
+				Source:         []string{"true"},
+				SourceExcludes: []string{"director"},
+			},
+			want: &opensearchapi.Response{
+				StatusCode: 200,
+				Header: http.Header{
+					"Content-Type": []string{"application/json; charset=UTF-8"},
+				},
+			},
+			wantBody: fmt.Sprintf(`{"_index":"%s","_id":"1","_version":2,"result":"noop","_shards":{"total":0,"successful":0,"failed":0},"_seq_no":2,"_primary_term":1,"get":{"_seq_no":2,"_primary_term":1,"found":true,"_source":{"title":"Moneyball","year":"2012"}}}`, index),
+			wantErr:  false,
+		},
+
+		// Bulk document
+		{
+			name: "TestBulkRequest_Do.",
+			r: opensearchapi.BulkRequest{
+				Index: index,
+				Body: strings.NewReader(`{ "index": { "_index": "movies", "_id": "tt1979320" } }
+{ "title": "Rush", "year": 2013 }
+{ "create": { "_index": "movies", "_id": "tt1392214" } }
+{ "title": "Prisoners", "year": 2013 }
+{ "update": { "_index": "movies", "_id": "tt0816711" } }
+{ "doc" : { "title": "World War Z" } }
+`),
+				Source: []string{"true"},
+			},
+			want: &opensearchapi.Response{
+				StatusCode: 200,
+				Header: http.Header{
+					"Content-Type": []string{"application/json; charset=UTF-8"},
+				},
+			},
+			wantErr: false,
+		},
+	}
+
+	client, err := opensearch.NewDefaultClient()
+	require.NoError(t, err)
+
+	iCreate := opensearchapi.IndicesCreateRequest{
+		Index:      index,
+		Pretty:     true,
+		Human:      true,
+		ErrorTrace: true,
+		Body:       strings.NewReader(fmt.Sprintf(`{"settings": {"index": {"number_of_shards": 4}}}`)),
+	}
+
+	iCreateResponse, err := iCreate.Do(context.Background(), client)
+	require.NoError(t, err)
+	require.Equalf(t, false, iCreateResponse.IsError(),
+		"Error when creating index template: %s", iCreateResponse.String())
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.r.Do(context.Background(), client)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Do() error = %+v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			require.Equalf(t, got.IsError(), tt.wantErr, "Do() got = %v, want %v", got.IsError(), tt.wantErr)
```

Comment:
Checking for `IsError()` is no longer needed .

Reply:
Fixing with next commit